<?xml version="1.0"?>
<launch>
    <!-- Gazebo options -->
    <arg name="use_rviz" default="true" />
    <arg name="benchmark" default="false" />
    <arg name="demo" default="false" />
    <arg name="random_shortcut_time" default="0.1" />
    <arg name="tight_shortcut" default="true" />
    <arg name="output_file" default="$(find mr_planner)/outputs/dual_gp4_benchmark.csv" />

    <!-- Start the robot state publisher, simulation, and controler -->
    <!-- <include file="$(find dual_gp4_moveit_config)/launch/demo_gazebo.launch">
    </include> -->

    <include file="$(find dual_gp4_moveit_config)/launch/demo.launch">
        <arg name="use_rviz" value="$(arg use_rviz)" />
    </include>

    <node name="left_arm" pkg="mr_planner" type="yk_robot" output="screen">
        <param name="group_name" value="left_arm" />
    </node>

    <node name="right_arm" pkg="mr_planner" type="yk_robot" output="screen">
        <param name="group_name" value="right_arm" />
    </node>

    <!-- Start the demo test node -->
    <node name="test_collision" pkg="mr_planner" type="demo" output="screen" required="true" if="$(arg demo)">
        <param name="num_robots" value="2" />
        <param name="movegroup_name" value="dual_arms" />
        <param name="group_name_0" value="left_arm" />
        <param name="group_name_1" value="right_arm" />
        <param name="mfi" value="false" />
        <param name="async" value="false" />
        <param name="shortcut" value="true" />
        <param name="tight_shortcut" value="true" />
        <param name="random_shortcut_time" value="0.1" />
        <param name="planner_name" value="RRTConnect" />
        <param name="planning_time_limit" value="4.0" />
        <param name="benchmark" value="false" />

        <param name="pose_name" value="left_push" />
    </node>

    <node name="test_collision" pkg="mr_planner" type="demo" output="screen" required="true" unless="$(arg demo)">
        <param name="num_robots" value="2" />
        <param name="movegroup_name" value="dual_arms" />
        <param name="group_name_0" value="left_arm" />
        <param name="group_name_1" value="right_arm" />
        <param name="mfi" value="false" />
        <param name="async" value="false" />
        <param name="shortcut" value="true" />
        <param name="random_shortcut_time" value="$(arg random_shortcut_time)" />
        <param name="tight_shortcut" value="$(arg tight_shortcut)" />
        <param name="planner_name" value="RRTConnect" />
        <param name="planning_time_limit" value="4.0" />
        <param name="benchmark" value="$(arg benchmark)" />
        <param name="output_file" value="$(arg output_file)" />

        <param name="pose_name" value="random1" />
        <param name="pose_name1" value="random2" />
        <param name="pose_name2" value="random3" />
        <param name="pose_name3" value="random4" />
        <param name="pose_name4" value="random5" />
        <param name="pose_name5" value="random6" />
        <param name="pose_name6" value="left_push_up" />
        <param name="pose_name7" value="right_push_up" />
        <param name="pose_name8" value="left_rotated" />
        <param name="pose_name9" value="right_rotated" />
        <param name="pose_name10" value="left_push" />
        <param name="pose_name11" value="right_push" />
        <param name="pose_name12" value="ready_pose" />
    </node>

</launch>